Exercises/SoerenHerrmann/exercises_07.py

```diff
@@ -0,0 +1,353 @@
+    # - 6: jump-if-false: if the first parameter is zero, it sets the instruction pointer to the value from the second parameter. Otherwise, it does nothing.
+    # - 7: less than: if the first parameter is less than the second parameter, it stores 1 in the position given by the third parameter. Otherwise, it stores 0.
+    # - 8: equals: if the first parameter is equal to the second parameter, it stores 1 in the position given by the third parameter. Otherwise, it stores 0.
+    # - 9: adjusts the relative offset by the value of its only parameter.
+    #      the offset increases by the value of the parameter (or decreases if the parameter value is negative).
+    # endregion
+    ADD = 1
+    MULTIPLY = 2
+    INPUT = 3
+    OUTPUT = 4
+    JUMP_IF_TRUE = 5
+    JUMP_IF_FALSE = 6
+    LESS_THAN = 7
+    EQUALS = 8
+    CHANGE_RELATIVE_OFFSET = 9
+    HALT = 99
+
+
+class tiles(enum.Enum):
+    EMPTY = 0
+    WALL = 1
+    BLOCK = 2
+    PADDLE = 3
+    BALL = 4
+
+# * My first programming language was Java, so there is always a voice in my head calling for classes, enums, etc.
+# * I am sorry for the overengineering, but I had to do it.
+# * I hope you can forgive me for this.
+
+class triplet():
+    def __init__(self, x: int, y: int, info: int) -> None:
+        self.x = int(x) 
+        self.y = int(y)
+        self.info = info
+
+
+placeholder = []
+score = 0
+
+# Initialize the plot
+fig, ax = plt.subplots()
+ax.set_aspect('equal')
+ax.set_xlim(43, 0)
+ax.set_ylim(30, 0)
+
+# Define colors for each tile type
+colors = {
+    tiles.EMPTY: 'black',
+    tiles.WALL: 'blue',
+    tiles.BLOCK: 'red',
+    tiles.PADDLE: 'green',
+    tiles.BALL: 'red'
+}
+
+
+def draw_tile(trplt: triplet):
+    global score
+    print(trplt.x, trplt.y, trplt.info)
+    if trplt.x == -1 and trplt.y == 0:
+        score = trplt.info
+        return
+    rect = plt.Rectangle((trplt.x, trplt.y), 1, 1, color=colors[tiles(trplt.info)])
+    ax.set_title(f'Score: {score}')
+    ax.add_patch(rect)
+    plt.draw()
+    plt.pause(0.00001)
+
+
+def get_element(mode: op_mode, dct: dict, idx: int, offset: int) -> int:
+    """returns element based upon the task mode"""
+    match mode:
+        case op_mode.POSITION:
+            return dct.get(dct.get(idx, 0), 0)
+        case op_mode.IMMEDIATE:
+            return dct.get(idx, 0)
+        case op_mode.RELATIVE:
+            return dct.get(dct.get(idx, 0) + offset, 0)
+
+
+def calc(dct: dict, idx: int, op: operations, instruction: list, offset: int) -> int:
+    """calc is short for calculator. Handles 1,2,7 and 8"""
+    param_one_mode = op_mode(int(instruction[0]))
+    param_two_mode = op_mode(int(instruction[1]))
+    idx += 1 
+    param_one = get_element(param_one_mode, dct, idx, offset)
+    idx += 1
+    param_two = get_element(param_two_mode, dct, idx, offset)
+    idx += 1
```

Comment:
idx += 1
This is redundant; please adjust it. My suggestion would be:
```suggestion
    param_one = get_element(param_one_mode, dct, idx + 1, offset)
    param_two = get_element(param_two_mode, dct, idx + 2, offset)
    idx += 3
```